nautobot/extras/models/metadata.py

```diff
@@ -0,0 +1,137 @@
+
+    class Meta:
+        ordering = ["name"]
+
+    def __str__(self):
+        return self.name
+
+    def clean(self):
+        super().clean()
+
+        if self.present_in_database:
+            # Check immutable fields
+            database_object = self.__class__.objects.get(pk=self.pk)
+            if self.data_type != database_object.data_type:
+                raise ValidationError({"data_type": "Type cannot be changed once created"})
+
+        # Choices can be set only on selection fields
+        if self.choices.exists() and self.data_type not in (
+            MetadataTypeDataTypeChoices.TYPE_SELECT,
+            MetadataTypeDataTypeChoices.TYPE_MULTISELECT,
+        ):
+            raise ValidationError("Choices may be set only for select/multi-select data_type.")
+
+
+@extras_features(
+    "custom_validators",
+    "graphql",
+    "webhooks",
+)
+class MetadataChoice(ChangeLoggedModel, BaseModel):
+    """Store the possible set of values for a select/multi-select metadata type."""
+
+    metadata_type = models.ForeignKey(
+        to=MetadataType,
+        on_delete=models.CASCADE,
+        related_name="choices",
+        limit_choices_to=models.Q(
+            data_type__in=[MetadataTypeDataTypeChoices.TYPE_SELECT, MetadataTypeDataTypeChoices.TYPE_MULTISELECT]
+        ),
+    )
+    value = models.CharField(max_length=CHARFIELD_MAX_LENGTH)
+    weight = models.PositiveSmallIntegerField(default=100, help_text="Higher weights appear later in the list")
+
+    documentation_static_path = "docs/user-guide/platform-functionality/metadata.html"
+
+    class Meta:
+        ordering = ["metadata_type", "weight", "value"]
+        unique_together = ["metadata_type", "value"]
+
+    def __str__(self):
+        return self.value
+
+    def clean(self):
+        super().clean()
+
+        if self.present_in_database:
+            # Check immutable fields
+            database_object = self.__class__.objects.get(pk=self.pk)
+            if self.metadata_type != database_object.metadata_type:
+                raise ValidationError({"metadata_type": "Cannot be changed once created"})
+
+        if self.metadata_type.data_type not in (
+            MetadataTypeDataTypeChoices.TYPE_SELECT,
+            MetadataTypeDataTypeChoices.TYPE_MULTISELECT,
+        ):
+            raise ValidationError(
+                {"metadata_type": "Metadata choices can only be assigned to select/multiselect data_type."}
+            )
+        # TODO: enforce validation_minimum, validation_maximum, validation_regex like CustomFieldChoice does?
+
+    def save(self, *args, **kwargs):
+        super().save(*args, **kwargs)
+        # TODO: on change of value, enqueue background task to go through all existing metadata and update them,
+        #       like CustomFieldChoice does?
+        # TODO: Alternately, should we let Metadata just have a FK directly to its corresponding MetadataTypeChoice?
```

Comment:
The downside of the existing pattern is data denormalization. If I have a MetadataChoice with value `"SNow"` and a bunch of Metadata using that choice, then when I change that choice value to `"ServiceNow"`:

- with the existing custom field pattern, we have to go through *each and every* Metadata using that choice and update it to match
- with a proper FK pattern, no other database changes are needed. 

It's necessary for custom fields because we can't store FKs in a JSONField, but I think it's a lot cleaner here to just use a FK. Regardless though, that's a decision we'll need to make when we implement the `Metadata` model, which is why it's a TODO for now.